generator/src/main/java/com/graphicsfuzz/generator/knownvaluegeneration/ExpressionGenerator.java

```diff
@@ -315,15 +323,53 @@ private Expr generateAdditionValue(Value value,
     if (value.getType() != BasicType.INT && value.getType() != BasicType.FLOAT) {
       return null;
     }
+    // If the given value is unknown, we are free to choose any arbitrary numbers for addition.
+    if (value.valueIsUnknown()) {
+      return new BinaryExpr(
+          generateExpr(factManager, currentFunction, stmtToInsertBefore,
+              new NumericValue((BasicType) value.getType(), Optional.empty())),
+          generateExpr(factManager, currentFunction, stmtToInsertBefore,
+              new NumericValue((BasicType) value.getType(), Optional.empty())),
+          BinOp.ADD);
+    }
+
+    Number summandA;
+    // Given the expected type, we have to retrieve all values from the fact manager and filter only
+    // the facts that are known to compute particular values.
+    final List<Value> knownValues = factManager.getValuesFromType(value.getType())
+        .stream().filter(item -> !item.valueIsUnknown()).collect(Collectors.toList());
+    boolean genSummandsFromKnownValues = generator.nextBoolean();
+    // If we are able to find any known values with the correct type, we then consider choosing
+    // summands based on such values. Otherwise, we will have to pick a random number to make an
+    // addition expression.
+    if (!knownValues.isEmpty() && genSummandsFromKnownValues) {
+      final NumericValue candidate =
+          (NumericValue) knownValues.get(generator.nextInt(knownValues.size()));
+      summandA = candidate.getValue().get();
+    } else {
+      if (value.getType() == BasicType.FLOAT) {
+        // As floating number is more sensitive than integer we thus have to pick a random number
+        // carefully. To do so, we have to limit a reasonably small range of numbers.
```

Comment:
Please file a GitHub issue for revisiting how to generate floating-point values here, and do a TODO(github issue URL) comment here.

And for now I suggest choosing a random integer in the range [-10, 10] (by getting an integer in [0-20] and subtracting 10 from it) and casting it to float, so that we at least get negative floats.